nixos-modules/microvm/store-disk.nix

```diff
@@ -1,9 +1,14 @@
 { config, lib, pkgs, ... }:
 
 let
-  regInfo = pkgs.closureInfo {
-    rootPaths = [ config.system.build.toplevel ];
-  };
+  regInfo = let
+    regInfo' = pkgs.closureInfo {
+      rootPaths = [ config.system.build.toplevel ];
+    };
+  in if config.microvm.storeOnDisk then
```

Comment:
I think it's the other way around: `unsafeDiscardStringContext` when `storeOnDisk == true` because then the store image contains everything.

Please also add a comment like: `# When a store disk is used, we can drop dependencies to the /nix/store because the squashfs/erofs contains all paths.`